foreshadow/intents/general.py

```diff
@@ -1,287 +0,0 @@
-    CategoricalEncoder,
-    MultiImputer,
-    Scaler,
-    SimpleImputer,
-    TextEncoder,
-)
-
-
-def _mode_freq(s, count=10):
-    """Compute the mode and the most frequent values.
-
-    Args:
-        s (:obj:`Series <pandas.Series>`): the series to analyze
-        count (int): the n number of most frequent values
-
-    Returns:
-        A tuple with the list of modes and (the 10 most common values,
-        their frequency counts, % frequencies)
-
-    """
-    mode = s.mode().values.tolist()
-    vc = s.value_counts().nlargest(count).reset_index()
-    vc["PCT"] = vc.iloc[:, -1] / s.size
-    return (mode, vc.values.tolist())
-
-
-def _outliers(s, count=10):
-    """Compute the mode and the most frequent values.
-
-    Args:
-        s (:obj:`Series <pandas.Series>`): the series to analyze
-        count (int): the n largest (magnitude) outliers
-
-    Returns:
-        a :obj:`Series <pandas.Series>` of outliers
-
-    """
-    out_ser = s[np.abs(s - s.mean()) > (3 * s.std())]
-    out_df = out_ser.to_frame()
-    out_df["selector"] = out_ser.abs()
-
-    return out_df.loc[out_df["selector"].nlargest(count).index].iloc[:, 0]
-
-
-def _standard_col_summary(df):
-    data = df.iloc[:, 0]
-    nan_num = int(data.isnull().sum())
-    mode, top10 = _mode_freq(data)
-
-    return OrderedDict([("nan", nan_num), ("mode", mode), ("top10", top10)])
-
-
-class GenericIntent(BaseIntent):
-    """See base class.
-
-    Serves as root of Intent tree. In the case that no other intent applies
-    this intent will serve as a placeholder.
-
-    """
-
-    children = ["TextIntent", "NumericIntent", "CategoricalIntent"]
-    """Match to CategoricalIntent over NumericIntent"""
-
-    single_pipeline_template = []
-    """No transformers"""
-
-    multi_pipeline_template = [
-        PipelineTemplateEntry("multi_impute", MultiImputer, False)
-    ]
-    """Perform multi imputation over the entire DataFrame."""
-
-    @classmethod
-    def is_intent(cls, df):
-        """Return true by default such that a column must match this.
-
-        .. # noqa: I101
-        .. # noqa: I201
-
-        """
-        return True
-
-    @classmethod
-    def column_summary(cls, df):
-        """No statistics can be computed for a general column.
-
-        .. # noqa: I101
-        .. # noqa: I201
-
-        """
-        return {}
-
-
-class NumericIntent(GenericIntent):
-    """See base class.
-
-    Matches to features with numerical data.
-
-    """
-
-    children = []
-    """No children"""
-
-    single_pipeline_template = [
-        PipelineTemplateEntry("dropper", DropFeature, False),
-        PipelineTemplateEntry("simple_imputer", SimpleImputer, False),
-        PipelineTemplateEntry("scaler", Scaler, True),
-    ]
-    """Perform imputation and scaling using Smart Transformers"""
-
-    multi_pipeline_template = []
-    """No multi pipeline"""
-
-    @classmethod
-    def is_intent(cls, df):
-        """Return true if data is numeric according to pandas.
-
```

Comment:
Before deleting these files, we should make sure to snag the intent column summary functions

Reply:
old.